### PR TITLE
chore: Update to operator-rs 0.77.1 and use new S3 structs

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Part of https://github.com/stackabletech/issues/issues/640

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] CRD changes approved
- [x] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [x] Helm chart can be installed and deployed operator works
- [x] Integration tests passed (for non trivial changes)
- [x] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

**Comment by sbernauer** [comment]
Started https://testing.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/trino-operator-it-custom/14/

File: rust/crd/src/catalog/delta_lake.rs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 use super::commons::{HdfsConnection, MetastoreConnection};
 use serde::{Deserialize, Serialize};
 use stackable_operator::{
-    commons::s3::S3ConnectionDef,
+    commons::s3::S3ConnectionInlineOrReference,
```
Comment (NickLarsenNZ) [inline rust/crd/src/catalog/delta_lake.rs:4]: thought: The name of the struct is tied to the variants inside which are not guaranteed to change.

_I know this should have been brought up earlier. Nothing to change, but just communicating thought_.

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

File: rust/crd/src/catalog/delta_lake.rs (hunk 1 above)
Comment (sbernauer) [reply rust/crd/src/catalog/delta_lake.rs:4]: Makes sense, thanks!

File: Cargo.toml [hunk 2]
```diff
@@ -24,11 +24,12 @@ serde = { version = "1.0", features = ["derive"] }
...
 product-config = { git = "https://github.com/stackabletech/product-config.git", tag = "0.7.0" }
 strum = { version = "0.26", features = ["derive"] }
-tokio = { version = "1.39", features = ["full"] }
+tokio = { version = "1.40", features = ["full"] }
 tracing = "0.1"
 
-# [patch."https://github.com/stackabletech/operator-rs.git"]
-# stackable-operator = { git = "https://github.com/stackabletech//operator-rs.git", branch = "main" }
+[patch."https://github.com/stackabletech/operator-rs.git"]
+stackable-operator = { git = "https://github.com/stackabletech//operator-rs.git", branch = "fix/clashing-volumes-and-mounts-2" }
```
Comment (NickLarsenNZ) [inline Cargo.toml:34]: Blocking merge until https://github.com/stackabletech/operator-rs/pull/871 is merged.

```suggestion
# [patch."https://github.com/stackabletech/operator-rs.git"]
# stackable-operator = { git = "https://github.com/stackabletech//operator-rs.git", branch = "main" }
```

**Review by NickLarsenNZ — CHANGES_REQUESTED** [review]
Just blocking this until https://github.com/stackabletech/operator-rs/pull/871 is merged.

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

**Comment by NickLarsenNZ** [comment]
It could be worth a changelog entry

**Comment by sbernauer** [comment]
Good catch, added in 285e805

**Review by NickLarsenNZ — APPROVED** [review]

**Pull request merged**